### PR TITLE
Support locus predicate in Transform

### LINKED ISSUE #1047: Add generic ReferenceRegion pushdown for parquet files
Would look like this:

```
val predicate: FilterPredicate = ((LongColumn("end") >= region.start) && (LongColumn("start") <= region.end) && (BinaryColumn("contigName") === Binary.fromString(region.referenceName)))
```

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1047, #1539. H/T to @ryan-williams for the really useful `genomic-loci` library. One pending task:

- [x] Update `Transform` markdown docs.
- [x] Add `NucleotideContigFragment`

Code is ready for review.

### CODE REVIEW

**Comment by heuermh** [comment]
@ryan-williams Would adding this dependency complicate your builds too much?  E.g. if ADAM depends on genomic-loci that may peg any projects of yours downstream of ADAM to a version you don't like.

Locus/loci as terms aren't used very consistently across our API and docs, might region predicate be better?

**Comment by fnothaft** [comment]
This is good to go on my end.

**Comment by heuermh** [comment]
> Locus/loci as terms aren't used very consistently across our API and docs, might region predicate be better?

@fnothaft @devin-petersohn @akmorrow13 Thoughts on this question?

**Comment by fnothaft** [comment]
>> Locus/loci as terms aren't used very consistently across our API and docs, might region predicate be better?
>
> Thoughts on this question?

Either way is fine with me.

**Comment by fnothaft** [comment]
Ping for review...

**Comment by heuermh** [comment]
I'll vote for region predicate.  Haven't heard from @ryan-williams so I assume the circular dependency will be fine.  I looked at the dependency tree resulting from adding the dependency and it looks fine to me.

**Comment by fnothaft** [comment]
Rebased, and changed locus predicate to region predicate.

File: docs/source/50_cli.md [hunk 1]
```diff
@@ -146,6 +146,14 @@ These options fall into several general categories:
       not load the `attributes` or `origQual` fields of the `AlignmentRecord`.
     * `-aligned_read_predicate`: If loading as Parquet, only loads aligned
       reads.
+    * `-locus_predicate`: A string indicating that reads should be filtered on
```
Comment (heuermh) [inline docs/source/50_cli.md]: missed replace here...

File: docs/source/50_cli.md [hunk 2]
```diff
@@ -146,6 +146,14 @@ These options fall into several general categories:
...
+      overlapping a genomic position or range. This argument takes a comma
+      separated list, where each element in the list takes the form:
+      * `contig:pos` for a single genomic position, or
+      * `contig:start-end` for a genomic range with closed start and open end
+      E.g., `-locus_predicate 1:100,2:1000-2000` would filter all reads that
```
Comment (heuermh) [inline docs/source/50_cli.md]: ...and here

**Comment by fnothaft** [comment]
Fixed the doc nits.

**Comment by heuermh** [comment]
Jenkins, retest this please

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/TransformAlignmentsSuite.scala [hunk 3]
```diff
@@ -70,4 +70,24 @@ class TransformAlignmentsSuite extends ADAMFunSuite {
     assert(qualityScoreCounts(30) === 92899)
     assert(qualityScoreCounts(10) === 7101)
   }
+
+  sparkTest("run locus predicate") {
```
Comment (heuermh) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/TransformAlignmentsSuite.scala]: sorry, missed these before

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/TransformAlignmentsSuite.scala [hunk 4]
```diff
@@ -70,4 +70,24 @@ class TransformAlignmentsSuite extends ADAMFunSuite {
...
+    // alas, copy resource does not work here...
+    val inputPath = testFile("sorted.bam")
+    val outputPath1 = tmpFile("predicate.1.adam")
+    val outputPath2 = tmpFile("predicate.2.adam")
+    val outputPath3 = tmpFile("predicate.3.adam")
+    TransformAlignments(Array(inputPath, outputPath1,
+      "-locus_predicate", "1:0-200,chr2:0-1000",
```
Comment (heuermh) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/TransformAlignmentsSuite.scala]: `-locus_predicate` &rarr; `-region_predicate`, and below

**Comment by fnothaft** [comment]
Fixed tests and rebased @heuermh.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**